tinker_cookbook/tests/test_renderers.py

```diff
@@ -17,161 +17,354 @@
-    render_name = (
-        get_recommended_renderer_name(model_name)
-        if not model_name.startswith("openai")
-        else "gpt_oss_medium_reasoning"
-    )
-    cookbook_renderer = get_renderer(render_name, tokenizer, image_processor)
-    convo: list[Message] = [
+    This is the standard test case for supervised example testing.
+    """
+    return [
         {"role": "user", "content": "Hello, how are you?"},
         {"role": "assistant", "content": "I'm fine, thank you!"},
-        {"role": "user", "content": "What is the capital of France?"},
     ]
 
+
+def get_tool_call_conversation() -> list[Message]:
+    """Full tool use conversation with tool call and response.
+
+    Includes: user request -> assistant tool call -> tool response -> assistant final answer.
+    """
+    return [
+        {"role": "user", "content": "What's the weather in San Francisco?"},
+        {
+            "role": "assistant",
+            "content": "I'll check the weather for you.",
+            "tool_calls": [
+                ToolCall(
+                    function=ToolCall.FunctionBody(
+                        name="get_weather",
+                        arguments='{"location": "San Francisco"}',
+                    ),
+                    id="call_123",
+                )
+            ],
+        },
+        {
+            "role": "tool",
+            "content": '{"temperature": 72, "condition": "sunny"}',
+            "tool_call_id": "call_123",
+        },
+        {"role": "assistant", "content": "The weather in San Francisco is sunny with 72°F."},
+    ]
+
+
+# Conversation registry for parametrized tests
+# Maps conversation ID to (factory_function, description, requires_tools)
+CONVERSATION_REGISTRY: dict[str, tuple[callable, str, bool]] = {
+    "basic_3turn": (get_basic_3turn_conversation, "basic 3-turn conversation", False),
+    "basic_2turn": (get_basic_2turn_conversation, "basic 2-turn conversation", False),
+    "tool_sft": (get_tool_call_conversation, "tool use conversation", True),
+}
+
+
+# Models that support tool calling in their renderers
+TOOL_CAPABLE_MODELS = {
+    "Qwen/Qwen3-30B-A3B",
+    "Qwen/Qwen3-VL-30B-A3B-Instruct",
+    "meta-llama/Llama-3.2-1B-Instruct",
+    "deepseek-ai/DeepSeek-V3.1",
+    "moonshotai/Kimi-K2-Thinking",
+}
+
+
+# =============================================================================
+# HF Compatibility Tests (parametrized by model and conversation)
+# =============================================================================
+
+
+def _prepare_conversation_for_model(
+    model_name: str, convo: list[Message], is_generation: bool
+) -> tuple[list[Message], list[Message]]:
+    """Prepare conversation for both cookbook renderer and HF template comparison.
+
+    Unfortunately, aug_convo and hf_convo sometimes need to differ because HF templates
+    auto-add certain content that our renderers expect to be present in the input:
```

Comment:
Also point out that only certain Message fields are even used by the HF renderer. E.g. it doesn't know how to use the tool call fields. (Is that right?)